ocaml/xapi-idl/lib/observer_helpers.ml

```diff
@@ -0,0 +1,264 @@
+(*
+ * Copyright (c) Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Rpc
+open Idl
+
+module D = Debug.Make (struct let name = "observer_interface" end)
+
+open D
+
+let service_name = "observer"
+
+let queue_name = Xcp_service.common_prefix ^ service_name
+
+let default_sockets_dir = "/var/lib/xcp"
+
+let default_path = Filename.concat default_sockets_dir service_name
+
+let uri () = "file:" ^ default_path
```

Comment:
I'm not a fan; how about using the URI module? There is always some confusion around the `//` part in file URLs.